pennylane_lightning/core/src/simulators/lightning_tensor/cutn/CutnBase.hpp

```diff
@@ -0,0 +1,222 @@
+     *
+     * @return size_t
+     */
+    size_t getWorkSpaceMemorySize(cutensornetWorkspaceDescriptor_t &workDesc) {
+        int64_t worksize{0};
+
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetWorkspaceGetMemorySize(
+            /* const cutensornetHandle_t */ getCutnHandle(),
+            /* cutensornetWorkspaceDescriptor_t */ workDesc,
+            /* cutensornetWorksizePref_t */
+            CUTENSORNET_WORKSIZE_PREF_RECOMMENDED,
+            /* cutensornetMemspace_t*/ CUTENSORNET_MEMSPACE_DEVICE,
+            /* cutensornetWorkspaceKind_t */ CUTENSORNET_WORKSPACE_SCRATCH,
+            /*  int64_t * */ &worksize));
+
+        return worksize;
+    }
+
+    /**
+     * @brief Set the memory for a work space
+     *
+     * @param workDesc cutensornet work space descriptor
+     * @param scratchPtr Pointer to scratch memory
+     * @param worksize Memory size of a work space
+     */
+    void setWorkSpaceMemory(cutensornetWorkspaceDescriptor_t &workDesc,
+                            void *scratchPtr, int64_t &worksize) {
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetWorkspaceSetMemory(
+            /* const cutensornetHandle_t */ getCutnHandle(),
+            /* cutensornetWorkspaceDescriptor_t */ workDesc,
+            /* cutensornetMemspace_t*/ CUTENSORNET_MEMSPACE_DEVICE,
+            /* cutensornetWorkspaceKind_t */ CUTENSORNET_WORKSPACE_SCRATCH,
+            /* void *const */ scratchPtr,
+            /* int64_t */ worksize));
+    }
+
+    /**
+     * @brief Save quantumState information to data provided by a user
+     *
+     * @param extentsPtr Extents of each sites
+     * @param tensorPtr Pointer to tensors provided by a user
+     */
+    void computeState(std::vector<int64_t *> &extentsPtr,
+                      std::vector<void *> &tensorPtr) {
+        cutensornetWorkspaceDescriptor_t workDesc;
+        PL_CUTENSORNET_IS_SUCCESS(
+            cutensornetCreateWorkspaceDescriptor(getCutnHandle(), &workDesc));
+
+        // TODO we assign half (magic number is) of free memory size to the
+        // maximum memory usage.
+        const std::size_t scratchSize = cuUtil::getFreeMemorySize() / 2;
+
+        PL_CUTENSORNET_IS_SUCCESS(cutensornetStatePrepare(
+            /* const cutensornetHandle_t */ getCutnHandle(),
+            /* cutensornetState_t */ getQuantumState(),
+            /* size_t maxWorkspaceSizeDevice */ scratchSize,
+            /* cutensornetWorkspaceDescriptor_t */ workDesc,
+            /*  cudaStream_t unused in v24.03*/ 0x0));
+
+        int64_t worksize = getWorkSpaceMemorySize(workDesc);
+
+        // Ensure data is aligned by 256 bytes
+        worksize += int64_t{256} - worksize % int64_t{256};
+
+        PL_ABORT_IF(static_cast<std::size_t>(worksize) > scratchSize,
+                    "Insufficient workspace size on Device!");
+
+        const std::size_t d_scratch_length = worksize / sizeof(size_t);
+        DataBuffer<size_t, int> d_scratch(d_scratch_length, getDevTag(), true);
```

Comment:
Mix of `std::size_t` and `size_t`. Worth normalising on whichever is used throughout LQ/LG/LK

Reply:
Unfortunately, std::size_t and size_t is mixed across ecosystem. I will use `std::size_t` for better portability in the `cutensornet` backend.